x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -110,79 +139,151 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
+        new Map()
+      );
+      response = await this.baseClient.find<T>({
+        ...options,
+        typeToNamespacesMap: undefined, // if the user is fully authorized, use `undefined` as the typeToNamespacesMap to prevent privilege escalation
+        ...(status === 'partially_authorized' && { typeToNamespacesMap, type: '', namespaces: [] }), // the repository requires that `type` and `namespaces` must be empty if `typeToNamespacesMap` is defined
+      });
+    } catch (error) {
+      auditor.add(savedObjectReadEvent, { action: 'saved_object_find', error });
+      throw error;
+    }
+    response.saved_objects.forEach((object) =>
+      auditor.add(savedObjectReadEvent, { action: 'saved_object_find', object })
     );
-    const response = await this.baseClient.find<T>({
-      ...options,
-      typeToNamespacesMap: undefined, // if the user is fully authorized, use `undefined` as the typeToNamespacesMap to prevent privilege escalation
-      ...(status === 'partially_authorized' && { typeToNamespacesMap, type: '', namespaces: [] }), // the repository requires that `type` and `namespaces` must be empty if `typeToNamespacesMap` is defined
-    });
+
     return await this.redactSavedObjectsNamespaces(response);
   }
 
   public async bulkGet<T = unknown>(
     objects: SavedObjectsBulkGetObject[] = [],
     options: SavedObjectsBaseOptions = {}
   ) {
-    const args = { objects, options };
-    await this.ensureAuthorized(this.getUniqueObjectTypes(objects), 'bulk_get', options.namespace, {
-      args,
-    });
+    const auditor = await this.getScopedAuditor();
+
+    let response: SavedObjectsBulkResponse<T>;
+    try {
+      const args = { objects, options };
+      await this.ensureAuthorized(
+        this.getUniqueObjectTypes(objects),
+        'bulk_get',
+        options.namespace,
+        {
+          args,
+        }
+      );
+
+      response = await this.baseClient.bulkGet<T>(objects, options);
+    } catch (error) {
+      objects.forEach((object) =>
+        auditor.add(savedObjectReadEvent, { action: 'saved_object_bulk_get', object, error })
+      );
+      throw error;
+    }
+    objects.forEach((object) =>
+      auditor.add(savedObjectReadEvent, { action: 'saved_object_bulk_get', object })
+    );
 
-    const response = await this.baseClient.bulkGet<T>(objects, options);
     return await this.redactSavedObjectsNamespaces(response);
   }
 
   public async get<T = unknown>(type: string, id: string, options: SavedObjectsBaseOptions = {}) {
-    const args = { type, id, options };
-    await this.ensureAuthorized(type, 'get', options.namespace, { args });
+    const auditor = await this.getScopedAuditor();
+
+    let savedObject: SavedObject<T>;
+    try {
+      const args = { type, id, options };
+      await this.ensureAuthorized(type, 'get', options.namespace, { args });
+
+      savedObject = await this.baseClient.get<T>(type, id, options);
+    } catch (error) {
+      auditor.add(savedObjectReadEvent, {
```

Comment:
This will record non-audit related failures. For example, if the object is not found. I think we should move `savedObject = await this.baseClient.get<T>(type, id, options);` outside of this try/catch block so that it's behavior is consistent with the other operations